### PR TITLE
Ref build

### PR DESCRIPTION
**Pull request by wojciech-cichocki**

### CODE REVIEW

File: programs/invariant/src/referral.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
+pub mod referral {
+    use anchor_lang::prelude::Pubkey;
+    use std::{borrow::Borrow, str::FromStr};
+    pub fn contains(ref_owner: Pubkey) -> bool {
+        #[cfg(feature = "jupiter")]
+        {
+            let p_key_one = Pubkey::from_str(&"JUP3c2Uh3WA4Ng34tw6kPd2G4C5BB21Xo36Je1s32Ph");
```
Comment (wojciech-cichocki) [inline programs/invariant/src/referral.rs:7]: This naming sucks. Try like that:
```
let jup2 = Pubkey::from_str(&"JUP3c2Uh3WA4Ng34tw6kPd2G4C5BB21Xo36Je1s32Ph").unwrap();
```

File: programs/invariant/src/referral.rs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            } else if ref_owner.eq(&example_one) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+
+        #[cfg(feature = "none")]
+        {
+            println!("none");
```
Comment (wojciech-cichocki) [inline programs/invariant/src/referral.rs:22]: remove all added `println!`

File: programs/invariant/src/referral.rs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    use anchor_lang::prelude::Pubkey;
+    use std::{borrow::Borrow, str::FromStr};
+    pub fn contains(ref_owner: Pubkey) -> bool {
+        #[cfg(feature = "jupiter")]
+        {
+            let p_key_one = Pubkey::from_str(&"JUP3c2Uh3WA4Ng34tw6kPd2G4C5BB21Xo36Je1s32Ph");
+            let example_one = p_key_one.unwrap();
+            let p_key_two = Pubkey::from_str(&"JUP2jxvXaqu7NQY1GmNF4m1vodw12LVXYxbFL2uJvfo");
+            let example_two = p_key_two.unwrap();
+            if ref_owner.eq(&example_two) {
```
Comment (wojciech-cichocki) [inline programs/invariant/src/referral.rs:11]: This if looks obscure. one `if` only is needed

File: programs/invariant/src/referral.rs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        let p_key = Pubkey::from_str(&"7CKCpJWFRu1WAWfCvDkwyniP6JGpSyMf4Bkxk2U6v2Ej");
+        let example = contains(p_key.unwrap());
+        assert!(!example);
+    }
+
+    #[test]
+    #[cfg_attr(not(feature = "jupiter"), ignore)]
+    fn test_is_referral_for_jupiter() {
+        let p_key = Pubkey::from_str(&"7CKCpJWFRu1WAWfCvDkwyniP6JGpSyMf4Bkxk2U6v2Ej");
+        let example = contains(p_key.unwrap());
```
Comment (wojciech-cichocki) [inline programs/invariant/src/referral.rs:63]: This naming sucks again!

File: package.json [hunk 5]
```diff
@@ -45,7 +45,10 @@
     "test:position-change": "anchor test --skip-build tests-staker/position-change.spec.ts",
     "build:invariant": "cd sdk && npm run build",
     "build:staker": "cd staker-sdk && npm run build",
-    "build:all": "npm run build:invariant && npm run build:staker"
+    "build:all": "npm run build:invariant && npm run build:staker",
+    "program:jupiter": "anchor build -- --features \"jupiter\"",
```
Comment (wojciech-cichocki) [inline package.json:49]: all `program:*` scripts failed

File: programs/invariant/src/referral.rs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    #[cfg_attr(any(feature = "jupiter", feature = "all"), ignore)]
+    fn test_is_referral_for_none() {
+        let example = Pubkey::from_str(&"7CKCpJWFRu1WAWfCvDkwyniP6JGpSyMf4Bkxk2U6v2Ej").unwrap();
+        assert!(!contains(example));
+    }
+
+    #[test]
+    #[cfg_attr(not(feature = "jupiter"), ignore)]
+    fn test_is_referral_for_jupiter() {
+        let example = Pubkey::from_str(&"7CKCpJWFRu1WAWfCvDkwyniP6JGpSyMf4Bkxk2U6v2Ej").unwrap();
```
Comment (wojciech-cichocki) [inline programs/invariant/src/referral.rs:51]: please not such naming. You can name
`Pubkey` var as `pubkey` or `pkey`. 
contains result as `is contains`

File: package.json (hunk 5 above)
Comment (BKorzeniewska) [reply package.json:49]: so, should I delete them?

Comment (wojciech-cichocki) [reply package.json:49]: nope, I mean they should end successfully, not fail :)

**Pull request merged**